packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -470,7 +470,7 @@ function shouldRestoreFocus(scopeRef: ScopeRef) {
     scope = scope.parent;
   }
 
-  return true;
+  return scope?.scopeRef === scopeRef;
```

Comment:
This was to fix tooltip, since the `Overlay` component includes a FocusScope. It should not try to restore focus unless focus was actually in the scope to begin with (not the case for tooltips).